File: src/com/sterndu/util/Listener.java
package com.sterndu.util;

import java.util.function.Consumer;

public class Listener<T> {
	
	private Consumer<T> m;
	private T data = null;
	
	public Listener(Consumer<T> c) {
		m = c;
	}
	
	public void call(T nv) {
		data = nv;
		m.accept(nv);
	}
	
	public T getData() {
		return data;
	}

}

File: src/com/sterndu/util/Listener.kt
@file:JvmName("Listener")
package com.sterndu.util

import java.util.function.Consumer

class Listener<T>(private val m: Consumer<T>) {
	var data: T? = null
		private set

	fun call(nv: T) {
		data = nv
		m.accept(nv)
	}
}
